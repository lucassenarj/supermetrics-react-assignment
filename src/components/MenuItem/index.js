import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import "./index.scss";

function MenuItem({ path, text }) {
  return (
    <li className="menu-item">
      <Link to={path}>{ text }</Link>
    </li>
  );
};

MenuItem.propTypes = {
  path: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default MenuItem;
