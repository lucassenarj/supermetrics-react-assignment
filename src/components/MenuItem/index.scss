@import "./../../assets/colors.scss";

.menu-item {
  margin-left: 10px;
}

.menu-item a {
  display: block;
  padding: 10px;
  color: $link;
}