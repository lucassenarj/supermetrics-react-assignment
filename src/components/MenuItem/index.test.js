import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import MenuItem from "./";

const values = {
  path: "/dashboard",
  text: "Dashboard"
}

describe("Components", () => {
  describe("MenuItem", () => {
    it("renders correctly", () => {
      const { container } = render(
        <MemoryRouter>
          <MenuItem path={values.path} text={values.text} />
        </MemoryRouter>
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
