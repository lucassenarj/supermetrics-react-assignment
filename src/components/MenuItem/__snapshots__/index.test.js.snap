// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components MenuItem renders correctly 1`] = `
<li
  class="menu-item"
>
  <a
    href="/dashboard"
  >
    Dashboard
  </a>
</li>
`;
